core/services/offchainreporting/oracle.go

```diff
@@ -88,15 +92,23 @@ func (d jobSpawnerDelegate) ServicesForSpec(spec job.Spec) ([]job.Service, error
 		return nil, errors.Errorf("offchainreporting.jobSpawnerDelegate expects an *offchainreporting.OracleSpec, got %T", spec)
 	}
 
-	gasLimit := d.config.EthGasLimitDefault()
-	transmitter := NewTransmitter(d.db.DB(), concreteSpec.TransmitterAddress.Address(), gasLimit)
+	contractFilterer, err := offchainaggregator.NewOffchainAggregatorFilterer(concreteSpec.ContractAddress.Address(), d.ethClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
+	}
+
+	contractCaller, err := offchainaggregator.NewOffchainAggregatorCaller(concreteSpec.ContractAddress.Address(), d.ethClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+	}
 
-	ocrContract, err := NewOCRContract(
+	ocrContract, err := NewOCRContractConfigTracker(
 		concreteSpec.ContractAddress.Address(),
+		contractFilterer,
+		contractCaller,
 		d.ethClient,
 		d.logBroadcaster,
 		concreteSpec.JobID(),
-		transmitter,
```

Comment:
split these into two objects, one which needs a transmitter and one which doesn't